README.md

```diff
@@ -11,30 +11,57 @@
   * [Join the Gitter chat room](https://gitter.im/cocotb)
 
 ## Installation
+Cocotb can be installed by running `pip install cocotb`, and requires:
 
-Cocotb can be installed by running `pip install cocotb`.
+- Python 3.5+
+- A C++ compiler
+- An HDL simulator (such as [Icarus Verilog](http://iverilog.icarus.com/))
 
-## Quickstart
+For more details see [the documentation](https://cocotb.readthedocs.io/en/latest/quickstart.html#pre-requisites).
 
-    # Install pre-requisites (waveform viewer optional)
-    sudo yum install -y iverilog python-devel gtkwave
+## Usage
 
-    # Checkout git repositories
-    git clone https://github.com/cocotb/cocotb.git
+For a D flip-flop description:
 
-    # Install cocotb
-    pip install ./cocotb
+```verilog
+module dff (
+    output reg q,
+    input c, d,
+);
 
-    # Run the tests...
-    cd cocotb/examples/endian_swapper/tests
-    make
+always @(posedge clk) begin
+    q <= d;
+end
 
-    # View the waveform
-    gtkwave waveform.vcd
+endmodule
+```
 
+An example of a simple randomized cocotb testbench:
 
-## Tutorials and examples
+```python
+import random
+import cocotb
+from cocotb.clock import Clock
+from cocotb.triggers import FallingEdge
+
+@cocotb.test()
+async def test_simple_dff(dut):
+    """ Test that D propagates to Q """
+    cocotb.fork(Clock(dut.c, 10, units="us").start())  # Create a 10us period clock on port `c`
+
+    for i in range(10):
+        val = random.randint(0, 1)
+        dut.d <= val
+        await FallingEdge(dut.c)
+        assert dut.q == val, "output was incorrect on the {}th cycle".format(i)
+```
+
+For more information please see the [cocotb documentation](https://cocotb.readthedocs.io/) and [wiki](https://github.com/cocotb/cocotb/wiki).
+
+## Tutorials, examples and related projects
 
 * [Endian Swapper tutorial](https://cocotb.readthedocs.org/en/latest/endian_swapper.html)
 * [Ping using TUN/TAP tutorial](https://cocotb.readthedocs.org/en/latest/ping_tun_tap.html)
-* [OpenCores JPEG Encoder example](https://github.com/chiggs/oc_jpegencode/)
+* [Cocotb based USB 1.1 test suite for FPGA IP, with testbenches for a variety of open source USB cores](https://github.com/antmicro/usb-test-suite-build)
```

Comment:
Maybe just point to https://github.com/cocotb/cocotb/wiki/Further-Resources#libraries-and-examples (and structure that a bit better)?

Reply:
See below.